controller/handler_peer_ctrl/command.go

```diff
@@ -31,7 +31,7 @@ func newCommandHandler(controller *raft.Controller) channel.TypedReceiveHandler
 	poolConfig := goroutines.PoolConfig{
 		QueueSize:   uint32(controller.Config.CommandHandlerOptions.MaxQueueSize),
 		MinWorkers:  0,
-		MaxWorkers:  uint32(controller.Config.CommandHandlerOptions.MaxWorkers),
+		MaxWorkers:  1, // we should only have one thing apply entries, so they don't get applied out of order
```

Comment:
I'm not sure, should probably do some more digging. I don't know what happens if you apply things out of order. But since it's a journal and it goes off of the index, I suspect it will either cache out of order entries and then apply them in order, or throw out out of order entries and then re-request them later.

Reply:
Out of order events or missing events shouldn't be possible or else coherence is lost. 

I'm sure raft does something.